lib/Sema/ConstraintSystem.cpp

```diff
@@ -3147,25 +3147,60 @@ std::string swift::describeGenericType(ValueDecl *GP, bool includeName) {
 static bool diagnoseConflictingGenericArguments(ConstraintSystem &cs,
                                                 const SolutionDiff &diff,
                                                 ArrayRef<Solution> solutions) {
-  if (!diff.overloads.empty())
+  // Also attempt to detect and diagnose conflicting generic arguments across
+  // multiple solutions and overloads when all fixes are generic mismatches.
+  bool allGenericMismatch =
+      llvm::all_of(solutions, [](const Solution &solution) -> bool {
```

Comment:
Is there an existing test-case which utilizes this since new examples have to do with argument type mismatches and not generic parameters?

Reply:
Right, so we are not recording the argument mismatch because of unhandled `GenericArgument` on `repairFailures` for arg to param locator elt. But even if we try to lift this restriction and record the argument fix, what is happening is that when solving two distinct disjunctions are formed: `bind to [C2.Element]` and `arg conv [C2.Element]`, the former failing and recording the generic mismatch and later the argument mismatch. But the reason we end up always with all solutions generic mismatches when attempting to diagnose the ambiguity is that the later disjunction is always considered the worst. 
Solver output ...
```
(solving component #0
                          (attempting disjunction choice [$T52] bind [C2.Element] [deep equality] [[locator@0x11f82b990 [Call@file.swift:7:5 -> apply argument -> comparing call argument #0 to parameter #0]]];
                            (attempting fix [fix: fix generic argument mismatch] @ locator@0x11f82b990 [Call@file.swift:7:5 -> apply argument -> comparing call argument #0 to parameter #0])
                            (increasing score due to attempting to fix the source)
                            (found solution 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0)
                          )
                          (attempting disjunction choice [$T52] arg conv [C2.Element] [array-upcast] [[locator@0x11f82b990 [Call@file.swift:7:5 -> apply argument -> comparing call argument #0 to parameter #0]]];
                            (increasing score due to collection upcast conversion)
                            (attempting fix [fix: allow argument to parameter type conversion mismatch] @ locator@0x11f82b990 [Call@file.swift:7:5 -> apply argument -> comparing call argument #0 to parameter #0])
                            (increasing score due to attempting to fix the source)
                            (solution is worse than the best solution)
                            (failed constraint [$T52] arg conv [C2.Element] [array-upcast] [[locator@0x11f82b990 [Call@file.swift:7:5 -> apply argument -> comparing call argument #0 to parameter #0]]];)
                          )
                        finished component #0)
                          (composed solution 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0)
```
I'm trying to see if is possible to detect this and tweak the score in order to invert that and get the argument mismatch as the best solution.

Reply:
Seems like even if we manage to get all argument mismatch by tweaking the score a bit(also not sure if we can do that), we still fail to diagnose that in `diagnoseConflictingGenericArguments` because we have two distinct overloads on the overload diff 
```
(lldb) p diff.overloads.size()
(size_t) $2 = 2
(lldb) p (*(diff.overloads.begin())).locator->getAnchor().dump()
(unresolved_dot_expr type='<null>' field 'map' function_ref=single
  (declref_expr type='<null>' decl=main.(file).SR1377f(_:_:).c2@file.swift:6:58 function_ref=unapplied))
(lldb) p (*(diff.overloads.begin() + 1)).locator->getAnchor().dump()
(unresolved_dot_expr type='<null>' field 'map' function_ref=single
  (declref_expr type='<null>' decl=main.(file).SR1377f(_:_:).c1@/file.swift:6:48 function_ref=unapplied))(lldb) 
```

Reply:
For this case, I still can't found a better solution than this one